src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -553,7 +553,10 @@ class WorkspaceService(protected val userInfo: UserInfo, dataSource: DataSource,
   def listEntityTypes(workspaceName: WorkspaceName): Future[PerRequestMessage] =
     dataSource.inFutureTransaction(readLocks=Set(workspaceName)) { txn =>
       withWorkspaceContextAndPermissions(workspaceName, WorkspaceAccessLevels.Read, txn) { workspaceContext =>
-        Future.successful(RequestComplete(StatusCodes.OK, containerDAO.entityDAO.getEntityTypes(workspaceContext, txn).toSeq))
+        val typesWithCounts = containerDAO.entityDAO.getEntityTypes(workspaceContext, txn).toSeq.map { entityType =>
+          entityType -> containerDAO.entityDAO.list(workspaceContext, entityType, txn).size
```

Comment:
Veto. This will load all the entities into memory just in order to count them, which is unnecessary.

What you actually want is something closer to the [entity pipeline](https://github.com/broadinstitute/rawls/blob/develop/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala#L251), but instead of filtering down to the specific entity, just count all vertices attached to the entityType one.